pkg/kubelet/stats/cadvisor_stats_provider.go

```diff
@@ -137,12 +139,66 @@ func (p *cadvisorStatsProvider) ListPodStats() ([]statsapi.PodStats, error) {
 		if vstats, found := p.resourceAnalyzer.GetPodVolumeStats(podUID); found {
 			podStats.VolumeStats = vstats.Volumes
 		}
+		podStats.EphemeralStorage = calEphemeralStorage(podStats.Containers, podStats.VolumeStats, &rootFsInfo)
 		result = append(result, *podStats)
 	}
 
 	return result, nil
 }
 
+func calEphemeralStorage(containers []statsapi.ContainerStats, volumes []statsapi.VolumeStats, rootFsInfo *cadvisorapiv2.FsInfo) *statsapi.FsStats {
+	result := &statsapi.FsStats{
+		Time:           metav1.NewTime(rootFsInfo.Timestamp),
+		AvailableBytes: &rootFsInfo.Available,
+		CapacityBytes:  &rootFsInfo.Capacity,
+		InodesFree:     rootFsInfo.InodesFree,
+		Inodes:         rootFsInfo.Inodes,
+	}
+	if containers != nil {
```

Comment:
Since the `len(nil_slice)` is `0`, it's ok to range over a nil slice. See: https://play.golang.org/p/LHC5rnt4ai

Reply:
fixed